plugins/prometheus.py

```diff
@@ -0,0 +1,109 @@
+# -*- coding: utf-8 -*-
+from jinja2 import Template
+import json
+from matrix_client.api import MatrixRequestError
+from neb.engine import KeyValueStore, RoomContextStore
+from neb.plugins import Plugin, admin_only
+from Queue import PriorityQueue
+from threading import Thread
+
+
+import time
+import logging as log
+import random
+
+queue = PriorityQueue()
+
+
+class PrometheusPlugin(Plugin):
+    """Plugin for interacting with Prometheus.
+    """
+    name = "prometheus"
+
+    #Webhooks:
+        #    /neb/prometheus
+    TYPE_TRACK = "org.matrix.neb.plugin.prometheus.projects.tracking"
+
+    TRACKING = ["track", "tracking"]
+
+    def __init__(self, *args, **kwargs):
+        super(PrometheusPlugin, self).__init__(*args, **kwargs)
+        self.store = KeyValueStore("prometheus.json")
+        self.rooms = RoomContextStore(
+            [PrometheusPlugin.TYPE_TRACK]
+        )
+        self.queue_counter = 1L
+        self.consumer = MessageConsumer(self.matrix)
+        self.consumer.daemon = True
+        self.consumer.start()
+
+    def on_receive_message(self, info):
+        log.info("recv %s", info)
+        template = Template(self.store.get("message_template"))
+        for alert in info["alert"]:
```

Comment:
`info` is effectively raw untrusted JSON from some random server. We should be guarding accesses to keys. Something like `info.get("alert", [])` should be fine as a drop-in replacement.